src/puppetlabs/puppetdb/http/server.clj

```diff
@@ -30,9 +42,30 @@
     (format "The %s API has been retired; please use v4" version)
     404)))
 
-(def routes
-  (app
-   ["v4" &] {:any v4-app}
+(defn query-app
```

Comment:
Also from our conversation this morning it would seem that we're already headed in a direction which is a single endpoint handler for everything and dispatch for a path happens from that endpoint.